cg/apps/downsample/downsample.py

```diff
@@ -0,0 +1,169 @@
+        dry_run: bool = False,
+    ):
+        """Initialize the API."""
+        super().__init__(config)
+        self.config: CGConfig = config
+        self.status_db: Store = config.status_db
+        self.hk_api: HousekeeperAPI = config.housekeeper_api
+        self.sample_id: str = sample_id
+        self.number_of_reads: float = number_of_reads
+        self.case_id: str = case_id
+        self.dry_run: bool = dry_run
+        self.downsample_data: DownsampleData = self.get_meta_data()
+        LOG.info(f"Starting DownsampleAPI for sample {sample_id}.")
+
+    def get_meta_data(self) -> DownsampleData:
+        """Return the DownSampleData.
+        Raises:
+            DownsampleFailedError
+        """
+        try:
+            return DownsampleData(
+                status_db=self.status_db,
+                hk_api=self.housekeeper_api,
+                sample_id=self.sample_id,
+                number_of_reads=self.number_of_reads,
+                case_id=self.case_id,
+                out_dir=Path(self.config.downsample),
+            )
+        except Exception as error:
+            raise DownsampleFailedError(repr(error))
+
+    def add_downsampled_sample_entry_to_statusdb(self) -> Sample:
+        """Add a down sampled sample entry to StatusDB."""
+        downsampled_sample: Sample = self.downsample_data.downsampled_sample
+        if sample_exists_in_statusdb(
+            status_db=self.status_db, sample_id=downsampled_sample.internal_id
+        ):
+            raise ValueError(f"Sample {downsampled_sample.internal_id} already exists in StatusDB.")
+        LOG.info(
+            f"New downsampled sample created: {downsampled_sample.internal_id} from {downsampled_sample.from_sample}"
+            f"Application tag set to: {downsampled_sample.application_version.application.tag}"
+            f"Customer set to: {downsampled_sample.customer}"
+        )
+        if not self.dry_run:
+            self.status_db.session.add(downsampled_sample)
+            self.status_db.session.commit()
+            LOG.info(f"Added {downsampled_sample.name} to StatusDB.")
+            return downsampled_sample
+        return downsampled_sample
+
+    def add_downsampled_case_to_statusdb(self) -> Family | None:
+        """
+        Add a down sampled case entry to StatusDB.
+        """
+        downsampled_case: Family = self.downsample_data.downsampled_case
+        if case_exists_in_statusdb(status_db=self.status_db, case_name=downsampled_case.name):
+            LOG.info(f"Case with name {downsampled_case.name} already exists in StatusDB.")
+            return
+        if not self.dry_run:
+            self.status_db.session.add(downsampled_case)
+            self.status_db.session.commit()
+            LOG.info(f"New down sampled case created: {downsampled_case.internal_id}")
+            return downsampled_case
+        return downsampled_case
+
+    def _link_downsampled_sample_to_case(self, sample: Sample, case: Family) -> None:
```

Comment:
This method likely belongs on the store class? We should settle on a clearer structure in which layer what kind of logic belongs. Feel free to keep it here though

Reply:
The statusdb functiion is `relate_sample` which is used in here. This function extends that a bit by performing the database commit.
Any ideas on how to do it better? I was thinking of adding the commit to the existing function but that might break other stuff so I did it like it is now.